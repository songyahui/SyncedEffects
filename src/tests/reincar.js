"use hopstrict"

var hh = require("hiphop");

var prg =
<hh.module O>
  <hh.loop>
    <hh.local S>
      <hh.if S>
	<hh.emit O/>
      </hh.if>
      <hh.pause/>
      <hh.emit S/>
    </hh.local>
  </hh.loop>
</hh.module>

exports.prg = new hh.ReactiveMachine(prg, "reincar");
