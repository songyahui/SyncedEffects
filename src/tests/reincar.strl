module reincar:

output O;

loop
	signal S in
	       present S then emit O end;
	       pause;
	       emit S;
	end signal
end loop

end module