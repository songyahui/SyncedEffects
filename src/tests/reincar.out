reincar> ;
--- Output:
reincar> ;
--- Output:
reincar> ;
--- Output:
reincar> ;
--- Output:
